src/main/java/com/github/onsdigital/babbage/search/external/SearchClient.java

```diff
@@ -0,0 +1,134 @@
+        Map<String, Future<SearchResult>> futures = new HashMap<>();
+
+        for (ONSQuery query : queryList) {
+            ProxyONSQuery request = new ProxyONSQuery(query);
+            Future<SearchResult> future = SearchClientExecutorService.getInstance().submit(request);
+            futures.put(query.name(), future);
+        }
+
+        return processFutures(futures);
+    }
+
+    public static LinkedHashMap<String, SearchResult> search(HttpServletRequest request, String listType) throws Exception {
+        Map<String, Future<SearchResult>> futures = new HashMap<>();
+
+        final String searchTerm = extractSearchTerm(request);
+        final int page = extractPage(request);
+        final int pageSize = extractSize(request);
+        final SortBy sortBy = extractSortBy(request, ContentQuery.DEFAULT_SORT_BY);
+
+        ListType listTypeEnum = ListType.forString(listType);
+
+        final Set<TypeFilter> typeFilters = extractSelectedFilters(request, listTypeEnum.getTypeFilters(), false);
+
+        SearchType[] searchTypes;
+        if (!listTypeEnum.equals(ListType.ONS)) {
+            searchTypes = new SearchType[]{SearchType.CONTENT, SearchType.COUNTS};
+        } else {
+            // Submit content
+            searchTypes = SearchType.getBaseSearchTypes();
+        }
+
+        for (SearchType searchType : searchTypes) {
+            SearchQuery searchQuery;
+            switch (searchType) {
+                case CONTENT:
+                    searchQuery = new ContentQuery(searchTerm, listTypeEnum, page, pageSize, sortBy, typeFilters);
+                    break;
+                case COUNTS:
+                    searchQuery = new TypeCountsQuery(searchTerm, listTypeEnum);
+                    break;
+                case FEATURED:
+                    searchQuery = new FeaturedResultQuery(searchTerm, listTypeEnum);
+                    break;
+                default:
+                    throw new Exception(String.format("Unknown searchType: %s", searchType.getSearchType()));
+            }
+            // Submit concurrent requests
+            Future<SearchResult> future = SearchClientExecutorService.getInstance().submit(searchQuery);
+            futures.put(searchType.getResultKey(), future);
+        }
+
+        // Wait until complete
+        return processFutures(futures);
+    }
+
+    private static LinkedHashMap<String, SearchResult> processFutures(Map<String, Future<SearchResult>> futures) throws ExecutionException, InterruptedException {
+        // Collect results
+        LinkedHashMap<String, SearchResult> results = new LinkedHashMap<>();
+
+        for (String key : futures.keySet()) {
+            SearchResult result = futures.get(key).get();
+            results.put(key, result);
+        }
+
+        return results;
+    }
+
+    public static void stop() throws Exception {
+        client.stop();
+    }
+
+    static class Shutdown extends Thread {
```

Comment:
You've duplicated this class - consider putting it in its own file and reusing it, or you could implement `Runnable` provide the `Run()` method and add the shutdown logic to that.